bundles/io/org.eclipse.smarthome.io.encryption/src/main/java/org/eclipse/smarthome/io/crypto/internal/CryptoActivator.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.crypto.internal;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+
+/**
+ * Extension of the default OSGi bundle activator
+ *
+ * @author Pauli Anttila - Intial contribution
+ */
+public final class CryptoActivator implements BundleActivator {
+
+    private static byte[] encKey;
+
+    /**
+     * Called whenever the OSGi framework starts our bundle
+     */
+    @Override
+    public void start(BundleContext bc) throws Exception {
```

Comment:
How do you ensure that this bundle is started before all the other bundles that want to read their (potentially encrypted) configuration? There are some discussions in issue #1896 already, but it is not implemented (yet).